File: dragon_go_countdown/src/main/java/net/tevp/dragon_go_countdown/DragonLogin.java
package net.tevp.dragon_go_countdown;

import android.util.Log;
import org.apache.commons.io.IOUtils;
import org.apache.http.HttpResponse;
import org.apache.http.client.methods.HttpGet;
import org.apache.http.cookie.Cookie;
import org.apache.http.impl.client.DefaultHttpClient;

import java.io.IOException;

import static android.util.Xml.Encoding.UTF_8;

public class DragonLogin {
    public static LoginResult Login(String username, String password) throws IOException {
        DefaultHttpClient httpClient = new DefaultHttpClient();
        HttpGet request = new HttpGet("http://www.dragongoserver.net/login.php?quick_mode=1&userid=" + username + "&passwd=" + password);
        HttpResponse response = httpClient.execute(request);
        String content = IOUtils.toString(response.getEntity().getContent(), String.valueOf(UTF_8));
        LoginResult result = new LoginResult();
        Log.d("DragonLogin", content);
        if (content.contains("#Error")) {
            if (content.contains("wrong_userid")) {
                result.setStatus(LoginStatus.BAD_USERNAME);
            } else if (content.contains("wrong_password")) {
                result.setStatus(LoginStatus.BAD_PASSWORD);
            } else {
                result.setStatus(LoginStatus.OTHER_ERROR);
            }
        } else if (!content.contains("Ok")) {
            result.setStatus(LoginStatus.OTHER_ERROR);
        } else {
            for (Cookie cookie : httpClient.getCookieStore().getCookies()) {
                if (cookie.getName().contentEquals("cookie_sessioncode")) {
                    result.setSessionCode(cookie.getValue());
                    result.setExpiry(cookie.getExpiryDate());
                    result.setStatus(LoginStatus.SUCCESS);
                    break;
                }
            }
        }
        return result;
    }
}

File: dragon_go_countdown/src/main/java/net/tevp/dragon_go_countdown/authentication/DragonLogin.kt
package net.tevp.dragon_go_countdown.authentication

import android.util.Log
import android.util.Xml.Encoding.UTF_8
import org.apache.commons.io.IOUtils
import org.apache.http.client.methods.HttpGet
import org.apache.http.impl.client.DefaultHttpClient
import java.io.IOException

object DragonLogin {
    @Throws(IOException::class)
    fun Login(username: String, password: String): LoginResult {
        val httpClient = DefaultHttpClient()
        val request = HttpGet("http://www.dragongoserver.net/login.php?quick_mode=1&userid=$username&passwd=$password")
        val response = httpClient.execute(request)
        val content = IOUtils.toString(response.entity.content, UTF_8.toString())
        val result = LoginResult()
        Log.d("DragonLogin", content)
        if (content.contains("#Error")) {
            if (content.contains("wrong_userid")) {
                result.status = LoginStatus.BAD_USERNAME
            } else if (content.contains("wrong_password")) {
                result.status = LoginStatus.BAD_PASSWORD
            } else {
                result.status = LoginStatus.OTHER_ERROR
            }
        } else if (!content.contains("Ok")) {
            result.status = LoginStatus.OTHER_ERROR
        } else {
            for (cookie in httpClient.cookieStore.cookies) {
                if (cookie.name.contentEquals("cookie_sessioncode")) {
                    result.sessionCode = cookie.value
                    result.expiry = cookie.expiryDate
                    result.status = LoginStatus.SUCCESS
                    break
                }
            }
        }
        return result
    }
}